### ISSUE TITLE
Configuring DesiredCapabilities object for each run

### ISSUE THREAD

**Issue by rafaltruszkowski** [sitespeedio/sitespeed.io#734]
Prior to getting started with Sitespeed.io, I had set up a nice Selenium/WebDriver system for testing our website. Among other things, the system performs tests on mobile and desktop. However, WebDriver was never meant to be used for analyzing the performance of websites, which is why I started using Sitespeed.io. 

I know there is a flag I can use to use a mobile viewport size, but it's not the same as running on an actual mobile device (something that I do using Selenium Grid 2). As far as I understand, the existing flags for Sitespeed.io do not allow me to use a "mobile node" unless it is the only node registered with browser firefox (or chrome). I would love to be able to configure the DesiredCapabilities object to specify my own parameters that are sent to the hub, which then picks the correct node. Is this something I could tweak myself? That way I could still utilize Sitespeed.io library, but pass all of the tests to my real mobile devices. I do not see why it would not be possible since Sitespeed.io library just communicates with the Selenium Grid.

**Comment by soulgalore** [comment]
Hi @rafalSuntimes ,
that would be really cool. So we use https://github.com/tobli/browsertime to drive the browser (that uses Selenium). Can you give me an example of how the DesiredCapabilities config would look like? Ping @tobli 

Best
Peter

**Comment by rafalSuntimes** [comment]
@soulgalore @tobli 

I wrote all my previous tests in Java. It's very easy to connect a WebDriver object to Selenium Server. It's as simple as saying

    DesiredCapabilities capabilities = DesiredCapabilities.firefox();
    capabilities.setCapability("platform", Platform.MAC);
    WebDriver driver = new RemoteWebDriver(new URL("http://192.168.43.14:4444/wd/hub"), capabilities);

, where "192.168.43.14:4444" is the location of the Selenium Server. Of course, in this case, the DesiredCapabilities capabilities object has its properties set to match the node I want. Just very regular Selenium Grid 2 stuff going on here.

But that's only for testing on a desktop host. In order to test on real mobile devices, I also needed to use the Appium framework. On a high level, Appium is just like Selenium Server, but for mobile devices. If I want my tests to run on my Android device for example, I would just say this in code:

     DesiredCapabilities capabilities = DesiredCapabilities.android();
     capabilities.setCapability("platformName", "Android");
     capabilities.setCapability("browserName", "Chrome");
     capabilities.setCapability("deviceName", "name_of_my_android_device");
     WebDriver driver = new AppiumDriver(new URL("http://192.168.43.14:3456/wd/hub"), capabilities);

Instead of pointing to port 4444, I used 3456, which is where Appium Server is running. Depending on what the DesiredCapabilities capabilities contains, Appium Server will pick a correct device, just like Selenium Server would for desktop nodes.

So assuming I already have Selenium Server and Appium Server running, it all comes down to just setting the DesiredCapabilities object and providing ip:port for those servers.

Does that answer your question?
